marconi-chain-index/src/Marconi/ChainIndex/Indexers/AddressDatum.hs

```diff
@@ -95,7 +99,7 @@ data AddressDatumHandle = AddressDatumHandle
     , _addressDatumHandleDiskStore :: Int
     }
 
-type instance StorableMonad AddressDatumHandle = IO
+type instance StorableMonad AddressDatumHandle = ExceptT IndexerError IO
```

Comment:
:( This definitely adds some boilerplate everywhere with `lift`.

And it reminds me off an (opinionated) article about the `ExceptT e IO a` anti-pattern.

I guess the question I have: how do we catch other IO exception that are not explicitly mentioned by `IndexerError`?

Reply:
We don't, unless we try/catch them (which is exactly what I've done for the SQLite errors).
Changing signatures for `MonadIO m, MonadError IndexerError m` helps a bit, and that's what I've tried to do as much as I could, but we still have to `liftIO` function that lives in IO.

Not dealing with `ExceptT` and in the large, with both `MonadError` and `Control.Exception` is a motivation in many framework, the most common noe being `RIO` I guess.
It has other downside though, and usually comes with a strongly opinionated prelude.
The "new indexers" hide a bit of this complexity and gather almost all of it in its core, it's still messy, but less painful to handle in new indexers.

If you remember the discussion we had about error handling, that's why I said that being explicit would be a lot messier. I do agree that being explicit is more informative and avoid forgetting about error handling, but yes, it has a cost.

Reply:
:+1: 

I guess in any `IO` code caller, he would need to wrap the runtime exception into our custom error type. I agree it's a trade-off. In the end, I think I'm okay with going this route if the rest of the team agrees.

Reply:
At least, if we want to allow the development of other backend aside the SQLite one, it's the safer way to do so: the coordinator can't just adapt its behaviour to any kind of implicit error sent along the `IO` monad with `throw`.